mono/metadata/appdomain.c

```diff
@@ -447,21 +447,21 @@ void
 mono_context_init_checked (MonoDomain *domain, MonoError *error)
 {
 	MonoClass *klass;
-	MonoAppContext *context;
+	MonoAppContextHandle context;
 
 	error_init (error);
 	if (mono_runtime_get_no_exec ())
 		return;
 
 	klass = mono_class_load_from_name (mono_defaults.corlib, "System.Runtime.Remoting.Contexts", "Context");
-	context = (MonoAppContext *) mono_object_new_pinned (domain, klass, error);
+	context = MONO_HANDLE_CAST (MonoAppContext, mono_object_new_pinned_handle (domain, klass, error));
```

Comment:
I think `mono_context_init_checked` needs a `HANDLE_FUNCTION_ENTER`/`RETURN` pair.  See, for example, `mono_context_init` which will leak a handle.